src/main/java/se/sundsvall/intricdatacollector/datasource/confluence/ConfluenceWorker.java

```diff
@@ -0,0 +1,269 @@
+            var jsonDocument = jsonPathParser.parse(json);
+            // Extract the page data
+            var page = toPage(municipalityId, pageId, jsonDocument);
+
+            // Skip any blacklisted pages
+            if (isBlacklisted(page.pageId(), page.ancestorIds())) {
+                LOG.info("Unable to process the page {} as it is blacklisted or has a blacklisted ancestor (municipalityId: {})", pageId, municipalityId);
+
+                return null;
+            }
+
+            // Get the Intric group id by traversing up the page tree
+            var intricGroupId = getIntricGroupId(pageId, page.ancestorIds());
+            // If we don't have an Intric group id, we can't really do anything more - bail out
+            if (isNotBlank(intricGroupId)) {
+                LOG.info("The page {} was matched with the Intric group id {} (municipalityId: {})", pageId, intricGroupId, municipalityId);
+            } else {
+                LOG.info("The page {} couldn't be matched with any Intric group (municipalityId: {})", pageId, municipalityId);
+
+                return null;
+            }
+
+            return PageBuilder.from(page)
+                .withIntricGroupId(intricGroupId)
+                .build();
+        });
+    }
+
+    void insertPage(final String pageId) {
+        // Get the page
+        getPageFromConfluence(pageId).ifPresentOrElse(page -> {
+            // Add an info blob to Intric
+            var blobId = intricIntegration.addInfoBlob(page.intricGroupId(), page.title(), page.bodyAsText(), page.url());
+
+            // Save the page
+            dbIntegration.savePage(PageBuilder.from(page).withIntricBlobId(blobId).build());
+
+            LOG.info("The page {} has been inserted (municipalityId: {})", pageId, municipalityId);
+        }, () -> LOG.info("Unable to insert the page {} since it couldn't be found in Confluence (municipalityId: {})", pageId, municipalityId));
+    }
+
+    void updatePage(final String pageId) {
+        // Get the page
+        getPageFromConfluence(pageId).ifPresentOrElse(page -> {
+            dbIntegration.getBlobId(pageId, municipalityId).ifPresentOrElse(blobId -> {
+                // Update the info blob in Intric
+                var newBlobId = intricIntegration.updateInfoBlob(page.intricGroupId(), blobId, page.title(), page.bodyAsText(), page.url());
+                // Save (update) the page
+                dbIntegration.savePage(PageBuilder.from(page).withIntricBlobId(newBlobId).build());
+
+                LOG.info("The page {} has been updated (municipalityId: {})", pageId, municipalityId);
+            }, () -> LOG.info("Unable to update the page {} since no Intric blob id could be found (municipalityId: {})", pageId, municipalityId));
+        }, () -> LOG.info("Unable to update the page {} since it couldn't be found in Confluence (municipalityId: {})", pageId, municipalityId));
+    }
+
+    void deletePage(final String pageId) {
```

Comment:
I suppose this could lead to data inconsistency if page successfully deleted from Intric but fails to be deleted from DB?

Add `@Transactional` to method and switch so that DB entity i removed before it gets deleted in Intric.? Then if Intric integration throws exception, the DB transaction will be rolled back.